test-framework/junit5-component/src/test/java/io/quarkus/test/component/nested/NestedTest.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.test.component.nested;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import jakarta.inject.Inject;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.quarkus.test.InjectMock;
+import io.quarkus.test.component.QuarkusComponentTest;
+import io.quarkus.test.component.TestConfigProperty;
+import io.quarkus.test.component.beans.Charlie;
+import io.quarkus.test.component.beans.MyComponent;
+
+@QuarkusComponentTest
+@TestConfigProperty(key = "foo", value = "BAR")
+public class NestedTest {
+
+    @Inject
+    MyComponent myComponent;
+
+    @InjectMock
+    Charlie charlie;
```

Comment:
I'd add similar injected fields into the nested class and test them as well.
(which works fine, I just want to create test coverage for it :))

Reply:
Good idea. The reason why I didn't add them before was that I thought that having the tested components/mocks injected in the top-level class is the only reasonable use case for nested test classes I can see ;-).

Reply:
Done.